src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1606,7 +1606,29 @@ public void UpdateWithExtraWorkspaceViewInfo(ExtraWorkspaceViewInfo workspaceVie
                 this,
                 new PointEventArgs(new Point2D(X, Y)));
 
-            foreach (ExtraNodeViewInfo nodeViewInfo in workspaceViewInfo.NodeViews)
+            LoadNodes(workspaceViewInfo.NodeViews);
+
+            // NOTE: This is here to support early JSON graphs
+            LoadLegacyNotes(workspaceViewInfo.Notes);
+
+            LoadNotesFromAnnotations(workspaceViewInfo.Annotations);
+
+            // All notes must be loaded/created before annotations to be 
+            // included in a containing annotation
+            LoadAnnotations(workspaceViewInfo.Annotations);
```

Comment:
wondering whether you can merge these 3 functions into one. 
`LoadAnnotations(workspaceViewInfo.Annotations, workspaceViewInfo.Notes)`.
Inside that function,
`if(annotations != null) then check for  (nodes.count() == 0), if so, then parse notes else parse annotations`
`if(notes != null) parse notes`.

Reply:
These methods were previously all just code blocks in the UpdateWithExtraWorkspaceViewInfo function. It seemed to me that the function was getting too large and difficult to comprehend so I split the code out into different functions so it would be clearer what was going on. Note that the notes need to be fully loaded (both legacy and new) before annotations are loaded so that notes can be contained inside of annotations.

Reply:
I agree, but splitting the code seems redundant. You can always check for notes, then annotations. Testing that single method seems easier than testing all these methods.

Reply:
After re-examining the splitting of the one large function into separate parts I still believe that it is clearer this way, as well as being easier to debug and test. I do not think that combining loading of legacy notes and the new annotation style notes into a single function would make things clearer or easier to work with, as the two methods have completely different code paths (i.e. one is iterating through the notes collection and the other is iterating through the annotation collection). Loading of annotations needs to occur after all notes, both legacy and new style, have been loaded so it makes sense to me to have a separate function for annotations after loading notes.

Reply:
Sure @ColinDayOrg. Write some comments on those functions, especially the  notes has to get serialized first.